lib/identity_cache/query_api.rb

```diff
@@ -106,6 +106,22 @@ def prefetch_associations(associations, records)
         nil
       end
 
+      protected
+
+      def record_from_coder(coder) #:nodoc:
+        if coder
+          record = instantiate(coder[:attributes].dup)
+
+          coder[:associations].each do |name, value|
```

Comment:
so to be sure the `name` here is the name of the association NOT of the associated model.

Reply:
Correct.  That is why `reflect_on_association(name)` on the following line works